src/Common/src/Interop/OSX/Interop.EventStream.cs

```diff
@@ -142,25 +144,77 @@ internal enum FSEventStreamCreateFlags : uint
         /// <param name="runLoop">The RunLoop to attach the stream to</param>
         /// <param name="runLoopMode">The mode of the RunLoop; this should usually be kCFRunLoopDefaultMode. See the documentation for RunLoops for more info.</param>
         [DllImport(Interop.Libraries.CoreServicesLibrary)]
-        internal static extern void FSEventStreamScheduleWithRunLoop(
+        private static extern void FSEventStreamScheduleWithRunLoop(
             FSEventStreamRef    streamRef,
             CFRunLoopRef        runLoop,
             CFStringRef         runLoopMode);
 
         /// <summary>
+        /// Attaches an EventStream to a RunLoop so events can be received. This should usually be the current thread's RunLoop.
+        /// </summary>
+        /// <param name="streamRef">The stream to attach to the RunLoop</param>
+        /// <param name="runLoop">The RunLoop to attach the stream to</param>
+        /// <param name="runLoopMode">The mode of the RunLoop; this should usually be kCFRunLoopDefaultMode. See the documentation for RunLoops for more info.</param>
+        internal static void FSEventStreamScheduleWithRunLoop(
+            SafeEventStreamHandle streamRef,
+            CFRunLoopRef runLoop,
+            SafeCreateHandle runLoopMode)
+
+        {
+            FSEventStreamScheduleWithRunLoop(streamRef.DangerousGetHandle(), runLoop, runLoopMode.DangerousGetHandle());
```

Comment:
Gotcha, I didn't know the Marshaller would handle the interop correctly when declared with SafeHandles. I'll remove the extra thunks (some need to be kept to abstract away extra parameters that we don't use) and I'll change the PInvoke to use the SafeHandle classes